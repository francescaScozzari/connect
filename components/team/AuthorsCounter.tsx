import React from 'react'
import { styled } from 'styled-components'
import { IconGroup } from '@/components/commons/Icons'

const AuthorsCounter = ({ counter }: { counter: number }) => {
  return (
    <Label>
      <IconGroup title="group" />
      <Bold as="span">{counter}</Bold> researchers identified
    </Label>
  )
}

const Label = styled.span`
  gap: 1em;
  font-size: 0.875rem;
  color: white;
`

const Bold = styled(Label)`
  margin-left: 0.5em;
  font-weight: 700;
`

export { AuthorsCounter }
