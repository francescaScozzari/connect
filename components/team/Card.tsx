import React from 'react'
import * as Accordion from '@radix-ui/react-accordion'

import { AccordionHeader } from './accordion/AccordionHeader'
import { AccordionContent } from './accordion/AccordionContent'

const Card = () => {
  return (
    <Accordion.Root type="single" defaultValue="item-1" collapsible>
      <Accordion.Item value="item-1">
        <AccordionHeader />
        <AccordionContent>
          Yes. It adheres to the WAI-ARIA design pattern.
        </AccordionContent>
      </Accordion.Item>

      <Accordion.Item value="item-2">
        <AccordionHeader>Is it unstyled?</AccordionHeader>
        <AccordionContent>
          Yes. It's unstyled by default, giving you freedom over the look and
          feel.
        </AccordionContent>
      </Accordion.Item>

      <Accordion.Item value="item-3">
        <AccordionHeader>Is it unstyled?</AccordionHeader>
        <AccordionContent>
          Yes. It's unstyled by default, giving you freedom over the look and
          feel.
        </AccordionContent>
      </Accordion.Item>

      <Accordion.Item value="item-4">
        <AccordionHeader>Is it unstyled?</AccordionHeader>
        <AccordionContent>
          Yes. It's unstyled by default, giving you freedom over the look and
          feel.
        </AccordionContent>
      </Accordion.Item>

      <Accordion.Item value="item-5">
        <AccordionHeader>Is it unstyled?</AccordionHeader>
        <AccordionContent>
          Yes. It's unstyled by default, giving you freedom over the look and
          feel.
        </AccordionContent>
      </Accordion.Item>

      <Accordion.Item value="item-6">
        <AccordionHeader>Is it unstyled?</AccordionHeader>
        <AccordionContent>
          Yes. It's unstyled by default, giving you freedom over the look and
          feel.
        </AccordionContent>
      </Accordion.Item>

      <Accordion.Item value="item-7">
        <AccordionHeader>Is it unstyled?</AccordionHeader>
        <AccordionContent>
          Yes. It's unstyled by default, giving you freedom over the look and
          feel.
        </AccordionContent>
      </Accordion.Item>
    </Accordion.Root>
  )
}

export { Card }
