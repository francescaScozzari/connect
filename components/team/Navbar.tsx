import React from 'react'
import styled from 'styled-components'
import Link from 'next/link'

import { SmallLogo } from '@/components/commons/Logo'
import { BackButton } from '@/components//commons/BackButton'

const Navbar = () => {
  return (
    <Nav>
      <Link href="/">
        <SmallLogo title="BI4E" />
      </Link>

      <Link href="/">
        <BackButton />
      </Link>
    </Nav>
  )
}

const Nav = styled.nav`
  width: 100%;
  background-color: ${({ theme }) => theme.colors.primary[0]};
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  padding: 2em 3em;
  border-bottom: 1px solid #c9d2ec;

  div {
    margin: 0 auto;
  }

  @media (max-width: 768px) {
    div {
      display: none;
    }
  }
`

export { Navbar }
