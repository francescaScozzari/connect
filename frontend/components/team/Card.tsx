import React from 'react'
import * as Accordion from '@radix-ui/react-accordion'

import { AccordionHeader } from './accordion/AccordionHeader'
import { AccordionContent } from './accordion/AccordionContent'
import { Author } from '@/models/Authors'

type CardProps = {
  authors: Author[]
}

const Card = ({ authors }: CardProps) => {
  return (
    <Accordion.Root type="single" defaultValue="item-0" collapsible>
      {authors.map((author, index) => {
        return (
          <Accordion.Item key={index} value={`item-${index}`}>
            <AccordionHeader
              fullName={author.fullName}
              university={author.university}
              orcid={author?.orcid}
            />
            <AccordionContent documents={author.documents} />
          </Accordion.Item>
        )
      })}
    </Accordion.Root>
  )
}

export { Card }
