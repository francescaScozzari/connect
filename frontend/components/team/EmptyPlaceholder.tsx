import React from 'react'
import { styled } from 'styled-components'
import Link from 'next/link'

import { IconNoGroup } from '@/components/commons/Icons'
import { H2 } from '../commons/Typography'
import { BackButton } from '../commons/BackButton'
import { SearchTips } from '../commons/SearchTips'
import { AuthorsCounter } from './AuthorsCounter'

const EmptyPlaceholder = () => {
  return (
    <MainSection>
      <AuthorsCounter counter={0} />
      <Container>
        <IconNoGroup title="no team" />
        <H2.Normal color="#2B2D42">Your search returned no results</H2.Normal>
        <Link href="/">
          <BackButton />
        </Link>
        <SearchTips />
      </Container>
    </MainSection>
  )
}

const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  gap: 2em;
  padding: 4em;
  border-radius: 20px;
  background-color: white;
`

const MainSection = styled.section`
  width: 100%;
  padding: 4em;
  display: flex;
  flex-direction: column;
  justify-content: start;
  gap: 1em;
  background-color: ${({ theme }) => theme.colors.primary[0]};
`

export { EmptyPlaceholder }
