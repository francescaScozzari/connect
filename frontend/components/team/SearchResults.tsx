import React from 'react'
import { styled } from 'styled-components'

import { Text } from '@/components/commons/Typography'
import { Card } from './Card'
import { AuthorsCounter } from './AuthorsCounter'
import type { Author } from '@/models/Authors'
import { highlightSentence } from '@/utils/utils'

type Props = {
  authors: Author[]
  givenSentence: { text: string; highlights: string[] }
}

const SearchResults = ({ authors, givenSentence }: Props) => {
  const sentence =
    highlightSentence(givenSentence.text, givenSentence.highlights) ?? ''

  return (
    <MainSection>
      <Sidebar>
        <AuthorsCounter counter={authors.length} />
        <Paragraph>
          <Title>
            Why weâ€™re suggesting these researches for your research project
          </Title>
          <br />
          <span
            dangerouslySetInnerHTML={{
              __html: sentence
            }}
          ></span>
        </Paragraph>
      </Sidebar>
      <CardsContainer>
        <Card authors={authors} />
      </CardsContainer>
    </MainSection>
  )
}

const Title = styled.span`
  display: inline-block;
  color: ${({ theme }) => theme.colors.primary[0]};
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.5;
`

const Paragraph = styled(Text.Normal)`
  font-size: 1.125rem;
  line-height: 2.25;
  color: #242424;
  height: fit-content;
  max-height: 60vh;
  word-break: break-all;
  text-align: start;
  padding: 2em;
  overflow: auto;
  background-color: white;
  border-radius: 1.25em;
  border: 1px solid #c9d2ec;
`

const MainSection = styled.section`
  width: 100%;
  padding: 4em;
  display: grid;
  column-gap: 3em;
  grid-template-columns: 0.5fr 1fr;
  grid-auto-flow: column;
  background-color: ${({ theme }) => theme.colors.primary[0]};
`

const Sidebar = styled.div`
  grid-column: 1;
  display: flex;
  flex-direction: column;
  height: fit-content;
  gap: 1em;
  position: sticky;
  top: 2em;
`

const CardsContainer = styled.div`
  grid-column: 2;
  display: flex;
  flex-direction: column;
  gap: 1em;
  width: 100%;
`

export { SearchResults }
