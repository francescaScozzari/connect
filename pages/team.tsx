import React from 'react'
import styled from 'styled-components'
import type { NextPage } from 'next'

import { H1 } from '@/components/commons/Typography'
import { Navbar } from '@/components/team/Navbar'
import { useAppSelector } from '@/store'
import { SearchResults } from '@/components/team/SearchResults'
import { EmptyPlaceholder } from '@/components/team/EmptyPlaceholder'

const Team: NextPage = () => {
  const { authors, q } = useAppSelector(state => state.search)

  return (
    <>
      <Navbar />
      <Container>
        <TitleSection>
          <H1.Medium color="white">Your research team</H1.Medium>
        </TitleSection>
        {authors.length ? (
          <SearchResults authors={authors} q={q} />
        ) : (
          <EmptyPlaceholder />
        )}
      </Container>
    </>
  )
}

const TitleSection = styled.section`
  width: 100%;
  padding: 2.5em;
  background-color: ${({ theme }) => theme.colors.primary[0]};
`

const Container = styled.div`
  display: flex;
  min-height: calc(100vh - 80px);
  align-items: center;
  justify-content: start;
  flex-direction: column;
`

export default Team
