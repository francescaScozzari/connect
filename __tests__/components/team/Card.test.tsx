import React from 'react'
import { expect } from '@jest/globals'
import { screen } from '@testing-library/react'

import { Card } from '@/components/team/Card'
import { renderWithWrappers } from '@/__tests__/functions'

const setup = () =>
  renderWithWrappers(
    <Card />
  )

describe('Card component', () => {
  test('Card renders correctly', () => {
    const container = setup()
    expect(container.firstChild).toMatchSnapshot()
  })
})
