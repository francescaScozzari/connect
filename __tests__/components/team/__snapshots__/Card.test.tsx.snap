// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component Card renders correctly 1`] = `
.c2 {
  color: white;
  margin: 0;
  text-transform: none;
}

.c3 {
  font-size: 24px;
  font-weight: 700;
}

.c6 {
  color: white;
  margin: 0;
  text-transform: none;
}

.c16 {
  color: #242424;
  margin: 0;
  text-transform: none;
}

.c17 {
  font-size: 16px;
  font-weight: 400;
}

.c9 {
  font-size: 16px;
  font-weight: 700;
}

.c7 {
  font-size: 14px;
  font-weight: 600;
}

.c10 {
  padding: 0.375em 0.5em;
  background-color: #2B2D42;
  border-radius: 1.25em;
}

.c11 {
  text-transform: uppercase;
  font-size: normal;
  color: white;
}

.c5 {
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 0.25em;
}

.c8 {
  text-transform: uppercase;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  max-width: 38em;
}

.c4 {
  display: flex;
  flex-flow: nowrap;
  justify-content: space-between;
}

.c1 {
  flex-grow: 1;
  display: flex;
  flex-direction: column;
  align-content: start;
  gap: 1em;
}

.c0 {
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: 1.5em;
  padding: 1.5em;
  border: 1px solid #c9d2ec;
  border-top-left-radius: 1.25em;
  border-top-right-radius: 1.25em;
  background-color: #434660;
  margin-bottom: 0;
}

.c0[data-state='closed'] {
  border-radius: 1.25em;
}

.c12 {
  all: unset;
  padding: 0.375em;
  height: 1.375em;
  width: 1.375em;
  border: 2px solid #c9d2ec;
  border-radius: 50%;
}

.c12[data-state='open']>svg {
  transform: rotate(180deg);
}

.c15 {
  display: inline-block;
  color: #2B2D42;
  font-size: 1.25rem;
  font-weight: 600;
  line-height: 1.5;
}

.c18 {
  line-height: 2.25;
  text-align: start;
}

.c14 {
  display: flex;
  flex-direction: column;
  gap: 1em;
  padding: 2em;
  width: 100%;
  background-color: #F4F7F8;
  border-bottom: 1px solid #c9d2ec;
}

.c13 {
  display: flex;
  flex-direction: column;
  align-items: center;
  border: 1px solid #c9d2ec;
  border-top: 0;
  border-bottom: 0;
}

.c13[data-state='closed'] {
  border: 0;
}

@media (max-width: 1600px) {
  .c8 {
    max-width: 28em;
  }
}

@media (max-width: 1440px) {
  .c8 {
    max-width: 16em;
  }
}

<div
  data-orientation="vertical"
>
  <div
    data-orientation="vertical"
    data-state="open"
  >
    <h3
      class="c0"
      data-orientation="vertical"
      data-state="open"
    >
      <div
        class="c1"
      >
        <h3
          class="c2 c3"
          color="white"
        >
          Leonard Hofstadter
        </h3>
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            <svg
              fill="none"
              height="26"
              width="27"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                author
              </title>
              <g>
                <path
                  d="M13.5 11.917c1.44 0 4.333.726 4.333 2.166v.174A5.724 5.724 0 0 1 13.5 16.25a5.724 5.724 0 0 1-4.334-1.993v-.174c0-1.44 2.893-2.166 4.334-2.166Zm0-1.084a2.173 2.173 0 0 1-2.167-2.166c0-1.192.975-2.167 2.167-2.167 1.191 0 2.166.975 2.166 2.167a2.173 2.173 0 0 1-2.166 2.166Zm6.5.217c0-3.932-2.871-6.717-6.5-6.717-3.63 0-6.5 2.785-6.5 6.717 0 2.535 2.112 5.893 6.5 9.902 4.387-4.009 6.5-7.367 6.5-9.902Zm-6.5-8.883c4.55 0 8.666 3.488 8.666 8.883 0 3.597-2.892 7.854-8.666 12.783-5.774-4.929-8.667-9.186-8.667-12.783 0-5.395 4.117-8.883 8.667-8.883Z"
                  fill="#fff"
                />
              </g>
            </svg>
            <p
              class="c6 c7 c8"
              color="white"
            >
              Princeton University
            </p>
          </div>
          <p
            class="c6 c9 c10"
            color="white"
          >
            ORCID 
            <strong
              class="c11"
            >
              0000-0000-0000-0001
            </strong>
          </p>
        </div>
      </div>
      <button
        aria-controls="radix-:r1:"
        aria-expanded="true"
        class="c12"
        data-orientation="vertical"
        data-radix-collection-item=""
        data-state="open"
        id="radix-:r0:"
        type="button"
      >
        <svg
          fill="none"
          height="24"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="m7.91 15.41 4.59-4.58 4.59 4.58L18.5 14l-6-6-6 6 1.41 1.41Z"
              fill="#C9D2EC"
            />
          </g>
        </svg>
      </button>
    </h3>
    <div
      aria-labelledby="radix-:r0:"
      class="c13"
      data-orientation="vertical"
      data-state="open"
      id="radix-:r1:"
      role="region"
      style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width); transition-duration: 0s; animation-name: none;"
    >
      <div
        class="c14"
      >
        <span
          class="c15"
        >
          test title
        </span>
        <p
          class="c16 c17 c18"
          color="#242424"
        >
          test description
        </p>
      </div>
    </div>
  </div>
</div>
`;
